### PR TITLE
Removes non-primitive types from TabPage

### LINKED ISSUE #9793: Only primitives in TabPage
## Test plan

https://github.com/brave/browser-laptop/pull/9794#issue-239768821

----

**Describe the issue you encountered:**
Remove `tabPageFrames` from `mergeProps`.

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #9793

Auditors: @bsclifton @bridiver

Test Plan:
- create two tap pages
- open youtube video in both tab pages
- right click on a tab page and try to mute/unmute
- try to close the whole tab page

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/reducers/contextMenuReducer.js [hunk 1]
```diff
@@ -33,6 +41,45 @@ const validateState = function (state) {
...
+    }
+  }, {
+    label: locale.translation('closeTabPage'),
+    click: () => {
+      windowActions.closeFrames(tabPageFrames)
+    }
+  }]
+
+  const tabPageMenu = Menu.buildFromTemplate(template)
+  tabPageMenu.popup(getCurrentWindow())
```
Comment (bsclifton) [inline app/renderer/reducers/contextMenuReducer.js]: Would be great to have a unit test that ensures when `windowConstants.WINDOW_ON_TAB_PAGE_MENU` happens that `electron.remote.Menu.buildFromTemplate` fires and that `popup` is called on the instance that it returns 😄

File: app/renderer/reducers/contextMenuReducer.js [hunk 2]
```diff
@@ -33,6 +41,45 @@ const validateState = function (state) {
...
+  })
+}
+
+const onTabPageMenu = function (state, action) {
+  action = validateAction(action)
+  state = validateState(state)
+
+  const frames = frameStateUtil.getNonPinnedFrames(state) || Immutable.List()
+  const tabsPerPage = Number(getSetting(settings.TABS_PER_PAGE))
+  const tabPageFrames = frames.slice(action.get('index') * tabsPerPage, (action.get('index') * tabsPerPage) + tabsPerPage)
```
Comment (bsclifton) [inline app/renderer/reducers/contextMenuReducer.js]: Would be good to read `action.get('index')` into a variable and skip processing if:
- index is null /undefined
- index is out of bounds (negative value, over the max number of tab pages)

Comment (NejcZdovc) [inline app/renderer/reducers/contextMenuReducer.js]: done

File: app/renderer/reducers/contextMenuReducer.js (hunk 1 above)
Comment (NejcZdovc) [inline app/renderer/reducers/contextMenuReducer.js]: done

**Comment by NejcZdovc** [comment]
@bsclifton can we merged this one?

File: app/renderer/components/tabs/tabPage.js [hunk 3]
```diff
@@ -69,6 +70,15 @@ class TabPage extends React.Component {
     e.preventDefault()
   }
 
+  onContextMenu (e) {
+    e.stopPropagation()
+    windowActions.onTabPageMenu(this.props.index)
```
Comment (bridiver) [inline app/renderer/components/tabs/tabPage.js]: just a nit, but I think `windowActions. onTabPageContextMenu` would be better here

Comment (NejcZdovc) [inline app/renderer/components/tabs/tabPage.js]: done

**Pull request merged**